src/NLopt.jl

```diff
@@ -27,15 +27,15 @@ end
 
 function __init__()
     # cfunction returns a raw pointer, so it must be called at runtime
-    global const munge_callback_ptr = cfunction(munge_callback, Ptr{Void},
-                                                (Ptr{Void}, Ptr{Void}))
+    global const munge_callback_ptr = cfunction(munge_callback, Ptr{Nothing},
+                                                (Ptr{Nothing}, Ptr{Nothing}))
     global const nlopt_callback_wrapper_ptr =
         cfunction(nlopt_callback_wrapper,
-                  Cdouble, (Cuint, Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Void}))
+                  Cdouble, (Cuint, Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Nothing}))
     global const nlopt_vcallback_wrapper_ptr =
-        cfunction(nlopt_vcallback_wrapper, Void,
+        cfunction(nlopt_vcallback_wrapper, Nothing,
                   (Cuint, Ptr{Cdouble}, Cuint, Ptr{Cdouble}, Ptr{Cdouble},
-                   Ptr{Void}))
+                   Ptr{Nothing}))
```

Comment:
You should use `Ptr{Cvoid}` for these kinds of `ccall` arguments.

Reply:
OK, made that change. Once I'm done with these suggestions, I'll make another PR.